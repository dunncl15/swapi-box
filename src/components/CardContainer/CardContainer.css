.CardContainer {
  position: relative;
  top: -190px;
  width: 100%;
  padding: 0 2.5%;
  background-color: #fff;
  color: #000;
  overflow: auto;
}
