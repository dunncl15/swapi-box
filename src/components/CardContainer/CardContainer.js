import React from 'react';
import './CardContainer.css';
import PeopleCard from '../PeopleCards/PeopleCards.js'
import PlanetCard from '../PlanetCards/PlanetCards.js'
import VehicleCard from '../VehicleCards/VehicleCards.js'

const getPeople = (data) => {

  return data.map((person, i) => {
    return <PeopleCard
            key={i}
            name={person.name}
            homeworld={person.homeworld} species={person.species}
            population={person.population_of_homeworld} />
  })
}

const getVehicle = (data) => {

  return data.map((vehicle, i) => {
    console.log(vehicle.name)
    return <VehicleCard
            key={i}
            name={vehicle.name}
            model={vehicle.model}
            passenger={vehicle.passengers}
            classCar={vehicle.vehicle_class} />
  })
}

const getCards = (selected, selectedCategory) => {
  if (selected === 'people') {
    return getPeople(selectedCategory)
  } if (selected === 'vehicles') {
    return getVehicle(selectedCategory)
  }
}

const CardContainer = ({ selectedCategory, selected }) => {

  return (
    <div className="CardContainer">
      <button>Favorites</button>
      { getCards(selected, selectedCategory) }
    </div>
  )
}

export default CardContainer;
