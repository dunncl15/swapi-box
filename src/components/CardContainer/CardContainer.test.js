import React              from 'react';
import { mount, shallow } from 'enzyme';
import ReactDOM           from 'react-dom';
import renderer           from 'react-test-renderer';

import CardContainer      from './CardContainer';

describe('CardContainer', () => {

  it('Renders CardContainer div', () => {
    const wrapper = shallow(
      <CardContainer />
    );

    expect(wrapper.find('div').length).toEqual(1);
  });

  it('Renders CardContainer div', () => {
    const wrapper = shallow(
      <CardContainer />
    );

    expect(wrapper.find('div').length).toEqual(1);
  });

  it('renders a snapshot', () => {
    const tree = renderer.create(<CardContainer />).toJSON();
    expect(tree).toMatchSnapshot();
});


});
