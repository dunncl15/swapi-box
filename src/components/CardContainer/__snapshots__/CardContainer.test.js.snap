exports[`CardContainer renders a snapshot 1`] = `
<div
  className="CardContainer" />
`;
